rcl/src/rcl/event.c

```diff
@@ -0,0 +1,182 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/event.h"
+
+#include <stdio.h>
+
+#include "rcl/error_handling.h"
+#include "rcl/expand_topic_name.h"
+#include "rcl/remap.h"
+#include "rcutils/logging_macros.h"
+#include "rmw/error_handling.h"
+#include "rmw/validate_full_topic_name.h"
+#include "rmw/event.h"
+
+#include "./common.h"
+#include "./publisher_impl.h"
+#include "./subscription_impl.h"
+
+
+typedef struct rcl_event_impl_t
+{
+  rmw_event_t rmw_handle;
+  rcl_allocator_t allocator;
+} rcl_event_impl_t;
+
+
```

Comment:
nitpick: minimize vertical whitespace (https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace)

Here and below, I won't comment on each line since there are several.

Reply:
Removed